### DISCUSSION TITLE
[router-vite-plugin] Ability to prepend routeTree.gen.ts file content via an option.

### DISCUSSION THREAD

**Discussion by mgcrea** [TanStack/router#1100] [Ideas]
Would be nice to be able to easily prepend the content of the router file, eg.

Use case for me is to prepend `/* eslint-disable @typescript-eslint/no-explicit-any */` to pass eslint validation.

```
  plugins: [react(), TanStackRouterVite({ generatedRouteTree: "src/config/routeTree.gen.ts", prependRouteTree: "/* eslint-disable @typescript-eslint/no-explicit-any */\n" })],
```

**Comment by SeanCassiere** [comment]
You can ignore the routeTree.gen.ts file being linted entirely in your ESLint config.
```json
{
  "ignorePatterns": ["routeTree.gen.ts"],
}
```
Since this file is being generated by the **router-cli**, you can consider it to be "safe" and relies on not being interfered with for the types to work.
